### ISSUE TITLE
Crash if use VrVideoView inside ViewPager when user swiped few items.

### ISSUE THREAD

**Issue by dekalo-stanislav** [googlevr/gvr-android-sdk#202]
In my project we are VrVideoView inside ViewPager.
There are few videos that we want to display inside it.

However there are crash inside VrVideoView when user swiped few items.
See stacktrace below:
`08-11 13:25:52.707 12712 13524 E AndroidRuntime: FATAL EXCEPTION: GLThread 12902
08-11 13:25:52.707 12712 13524 E AndroidRuntime: Process: com.gfycat.android.debug, PID: 12712
08-11 13:25:52.707 12712 13524 E AndroidRuntime: java.lang.IllegalStateException: Unable to update texture contents (see logcat for details)
08-11 13:25:52.707 12712 13524 E AndroidRuntime: 	at android.graphics.SurfaceTexture.nativeUpdateTexImage(Native Method)
08-11 13:25:52.707 12712 13524 E AndroidRuntime: 	at android.graphics.SurfaceTexture.updateTexImage(SurfaceTexture.java:240)
08-11 13:25:52.707 12712 13524 E AndroidRuntime: 	at com.google.vr.sdk.widgets.video.VideoTexture.updateTexture(VideoTexture.java:71)
08-11 13:25:52.707 12712 13524 E AndroidRuntime: 	at com.google.vr.sdk.widgets.video.VrVideoPlayerInternal.prepareFrame(VrVideoPlayerInternal.java:344)
08-11 13:25:52.707 12712 13524 E AndroidRuntime: 	at com.google.vr.sdk.widgets.video.VrVideoRenderer.onDrawFrame(VrVideoRenderer.java:55)
08-11 13:25:52.707 12712 13524 E AndroidRuntime: 	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1535)
08-11 13:25:52.707 12712 13524 E AndroidRuntime: 	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1240)
`
This crash happens inside library without any calls from outside, right before it.

Would be very appreciated for any suggestions.

**Comment by digitalbuddha** [comment]
We are having same issue. We are putting a vrview inside of a recyclerview. There is only 1 videview which we keep moving to the "visible" viewholder.  Every so often (1 out of 5-10 times) the above error is thrown when we try to attach the view to a new parent.

**Comment by alexio** [comment]
Also having this issue. Injecting the videoview into the appropriate viewholder during `onAttachededToWindow` and removing it from the viewholder during `onDetachedFromWindow`. It will randomly crash when trying to insert/play the videoview when the viewholder is attached. Scrolling up and down will reproduce this issue. 

Only tested it on a Nexus 6P so far.

**Comment by digitalbuddha** [comment]
Here's a bit more info: the error is being thrown from a native method
```
static void SurfaceTexture_updateTexImage(JNIEnv* env, jobject thiz)
{
    sp<SurfaceTexture> surfaceTexture(SurfaceTexture_getSurfaceTexture(env, thiz));
    status_t err = surfaceTexture->updateTexImage();
    if (err == INVALID_OPERATION) {
        jniThrowException(env, IllegalStateException, "Unable to update texture contents (see "
                "logcat for details)");
    } else if (err < 0) {
        jniThrowRuntimeException(env, "Error during updateTexImage (see logcat for details)");
    }
}
```

which comes from java land here
```
 public synchronized void updateTexture() {
        if(this.needUpdateTexture) {
            if(this.surfaceTexture != null) {
                this.surfaceTexture.updateTexImage();
            }

            this.needUpdateTexture = false;
        }

    }
```

We see that we check whether the surfaceTexture is null which is great but I have a theory that the issue is that the view is currently detached.  Pure speculation but I bet if check whether the view is attached the crash will go away. 

Hope this helps :-)

**Comment by dekalo-stanislav** [comment]
@digitalbuddha, do you have access to google-vr lib source code? I did not saw any in public access.

**Comment by dav-cz** [comment]
Thanks for the reports. I was able to reproduce this and find a fix. It will be in our next release.

@digitalbuddha you were right the problem was that the view was just being detached, a last frame update was sent but the GL context was gone.

**Comment by dav-cz** [comment]
This was fixed in 1.0.0. 

(but we forgot to mention it in our release notes)
